diff --git a/Makefile b/Makefile
index bc77720..9831c2c 100644
--- a/Makefile
+++ b/Makefile
@@ -148,3 +148,3 @@
-DEBUG_CFLAGS = $(FULL_CFLAGS) -DDEBUG
-RELEASE_CFLAGS = $(FULL_CFLAGS) -O2
-COVERAGE_CFLAGS = $(FULL_CFLAGS) -O0 --coverage -fprofile-update=atomic
+DEBUG_CFLAGS = $(FULL_CFLAGS)
+RELEASE_CFLAGS = $(FULL_CFLAGS)
+COVERAGE_CFLAGS = $(FULL_CFLAGS) --coverage -fprofile-update=atomic
@@ -277 +277 @@
-	$(STRIP) $@
+#	$(STRIP) $@
